Source/Eto.Gtk/Forms/GtkWindow.cs

```diff
@@ -246,13 +249,11 @@ public override Size ClientSize
 			{
 				if (Control.IsRealized)
 				{
-					var diff = vbox.Allocation.Size.ToEto() - containerBox.Allocation.Size.ToEto();
-					Control.Resize(value.Width + diff.Width, value.Height + diff.Height);
-					SetMinMax(value + diff);
+					Control.Resize(value.Width, value.Height);
+					SetMinMax(value);
 				}
 				else
 				{
-					clientSize = value;
```

Comment:
Setting clientSize before control has allocated space wouldn't work so it has been set later on: https://github.com/k5jae/Eto/blob/322abd94a5e508e2e7f79c925e8d24ee9ee589b0/Source/Eto.Gtk/Forms/GtkWindow.cs#L266

With your changes Size and ClientSize would do the same thing before the control was realized.